tests/experimental/datasources/test_pandas_datasource.py

```diff
@@ -132,6 +146,23 @@ def test_data_asset_defined_for_io_read_method(self, method_name: str):
 
         assert type_name in asset_class_names
 
+    @pytest.mark.parametrize("asset_class", PandasDatasource.asset_types)
+    def test_minimal_validation(self, asset_class: Type[_FilesystemDataAsset]):
+        with pytest.raises(pydantic.ValidationError) as exc_info:
+            asset_class(  # type: ignore[call-arg] # type has default
+                name="test",
+                base_directory=pathlib.Path(__file__),
+                regex=re.compile(r"yellow_tripdata_sample_(\d{4})-(\d{2})"),
+                foo="bar",
+            )
+
+        errors_dict = exc_info.value.errors()
+        assert {
+            "loc": ("foo",),
+            "msg": "extra fields not permitted",
+            "type": "value_error.extra",
+        } == errors_dict[-1]
```

Comment:
Can we use `any` or `all` instead of indexing?

Reply:
I don't want to use `any` or `all` because I want to see the nice assert dict diff.
If the index is wrong I want it to fail.
There's one special case that prevents me from doing
```python
assert [{
            "loc": ("foo",),
            "msg": "extra fields not permitted",
            "type": "value_error.extra",
        }] == errors_dicts
```

Reply:
On second thought.... `kwargs` doesn't make sense as a required field. I'll make it optional and then this test can be simpler.

Reply:
I'd added comments to the test to explain.